File: Java/LeetCode/Reverse Words in a String.java
// https://leetcode.com/problems/reverse-words-in-a-string/

import java.util.StringJoiner;

class Solution {
    public String reverseWords(String s) {
        String[] words = s.split(" +");
        
        StringJoiner output = new StringJoiner(" ");

        for (int i = words.length-1; i >= 0; --i) {
            if (!words[i].isBlank()) {
                output.add(words[i]);
            }
        }

        return output.toString();
    }
}

File: Kotlin/LeetCode/Reverse Words in a String.kt
// https://leetcode.com/problems/reverse-words-in-a-string/

import java.util.StringJoiner;

class Solution {

    fun reverseWords(s: String): String {
        val words = s.trim().split(" +".toRegex());
        val output = StringJoiner(" ");

        for (i in words.size-1 downTo 0) {
            output.add(words[i]);
        }

        return output.toString();
    }

    // Alternative solution
    /*
    fun reverseWords(s: String): String {
        return s.trim().split(" +".toRegex()).asReversed().joinToString(" ");
    }
    */
    
}
